src/io/github/sspanak/tt9/ime/TraditionalT9.java

```diff
@@ -682,4 +677,21 @@ private void restoreAddedWordIfAny() {
 	protected View createSoftKeyView() {
 		return softKeyHandler.getView();
 	}
+
+
+	/**
+	 * forceShowWindowIfHidden
+	 * Some applications may hide our window and it remains invisible until the screen is touched or OK is pressed.
+	 * This is fine for touchscreen keyboards, but the hardware keyboard allows typing even when the window and the suggestions
+	 * are invisible. This function forces the InputMethodManager to show our window.
+	 */
+	protected void forceShowWindowIfHidden() {
+		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P
```

Comment:
Consider requesting focus and supporting older devices. More info: https://developer.android.com/develop/ui/views/touch-and-input/keyboard-input/visibility

Reply:
The example does not work without a view, but here we cannot get any View. Let's just ignore it on older devices.